Assets/Modules/Movement/Scripts/PlayerMovementController.cs

```diff
@@ -0,0 +1,95 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.InputSystem;
+
+public class PlayerMovementController : MonoBehaviour
+{
+
+    [SerializeField] private float jumpForce = 350f;
+
+    private bool gamepadIsConnected = false;
+
+    private Camera mainCamera;
+    private Rigidbody2D rb;
+    private MyInputActions characterActions;
+
+    private Vector2 aim;
+    private bool canJump = true;
+
+
+    [SerializeField] GameObject testerColorBOX;
+    [SerializeField] Color on;
+    [SerializeField] Color off;
+    private Color boxColor;
+
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        rb = GetComponent<Rigidbody2D>();
+        mainCamera = Camera.main;
+        characterActions = new MyInputActions();
+        characterActions.Enable();
+
+        //this asks to run this method OnDeviceChange when the device changes. Interesting concept.
+        InputSystem.onDeviceChange += OnDeviceChange;
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        // I'm trying to test if gamepadIsConnected is actually changing, but it doesn't look like it is.
+        if(!gamepadIsConnected)
+        {
+            // UpdateAimWithMouse();
+            testerColorBOX.GetComponent<SpriteRenderer>().color = off;
+        }
+        else{
+            testerColorBOX.GetComponent<SpriteRenderer>().color = on;
+        }
+    }
+
+    private void OnDeviceChange(InputDevice device, InputDeviceChange change)
+    {
+        // Update input devices when a device is added or removed
+        if (change == InputDeviceChange.Added || change == InputDeviceChange.Removed)
+        {
+            gamepadIsConnected = device is Gamepad; //same as if(device is Gamepad){gamepadIsConnected = true;}
```

Comment:
I'm not 100% sure but I think here you should check if the device is `Joystick` If you wanna check the controller 🤔 try it and see if it solves the input issue. If it didn't work I'll check it myself.

Reply:
Changing it to joystick does not solve the issue for me, the issue is that the mouse is still affecting the jump, even if I'm using the controller (even if I'm constantly moving the stick of the controller, which assumes to me the value of the stick is always changing, thus, the program should always detect it) but from time to time it still follows the mouse.
So Gamepad or Joystick both do the same and have the same issue unfortunately!